Controllers

rae-stanton opened the pull request:
Working on controllers for recipe and ingredients - WIP

rae-stanton committed:
WIP controllers

working on controllers for ingredients and recipes

rae-stanton committed:
controllers wip

working on controllers

rae-stanton commented:
@spaulsandhu

spaulsandhu commented on app/controllers/recipe_controller.rb:
```diff
@@ -1,2 +1,37 @@
 class RecipeController < ApplicationController
+  def new
+    @recipe = Recipe.new
+    @ingredients = Ingredient.all
```
Why are we setting the instance variable here for `all` Ingredients?

Would something like `@ingredients = @recipe.ingredients` make more sense so the ingredients we're instantiating for the view are scoped to the specific recipe we're viewing? 🤔

rae-stanton replied:
^ yeah that makes way more sense

spaulsandhu commented on app/controllers/recipe_controller.rb:
```diff
@@ -1,2 +1,37 @@
+
+  def edit
+    @recipe = Recipe.find(params[:id])
+    @ingredients = Ingredient.all
```
Same as above, do we want to set the instance variable to `ALL` Ingredients in the db or just the ingredients associated with the recipe?

spaulsandhu commented on test/controllers/ingredient_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class IngredientControllerTest < ActionDispatch::IntegrationTest
```
Do we plan on using minitest for this app or Rspec?

rae-stanton replied:
Going to add rspec for test suite~

spaulsandhu commented on test/controllers/recipe_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class RecipeControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Rails generates spec files for minitest when using generators, so we may want to re-assess adding these now if we plan on using `Rspec` for our testing purposes.

spaulsandhu replied:
We should delete these files in a different PR.

spaulsandhu reviewed (commented):
This looks like a good start, but I think we have some clean-up and reorganizing of some of our logic to take care of before we can merge down. Lemme know if you want to chat about things or if my comments aren't clear and I'll be happy to go over details with you. 👍🏽

rae-stanton commented:
Leaving this open until rspec has been added versus minitest generated tests

rae-stanton committed:
Made changes for author user type

Changed recipe to have an author/author ID versus just using `user` - used this guide to assist.

https://www.theodinproject.com/lessons/ruby-on-rails-active-record-associations#basic-stuff

Tested in console, writing tests via rspec in future commit.

spaulsandhu reviewed (approved):
LGTM
